packages/vite-plugin-svelte/src/utils/options.ts

```diff
@@ -177,6 +175,23 @@ function enforceOptionsForProduction(options: ResolvedOptions) {
 	}
 }
 
+function removeIgnoredOptions(options: ResolvedOptions) {
+	const ignoredCompilerOptions = ['generate', 'cssHash', 'format', 'filename'];
+	const passedCompilerOptions = Object.keys(options.compilerOptions || {});
+	const passedIgnored = passedCompilerOptions.filter((o) => ignoredCompilerOptions.includes(o));
+	if (passedIgnored.length) {
+		log.warn(
+			`The following Svelte compilerOptions are controlled by vite-plugin-svelte and essential to its functionality. User-specified values are ignored. Please remove them from your configuration: ${passedIgnored.join(
+				', '
```

Comment:
Was initially thinking to remove the props from the `CompilerOptions` types directly, but maybe we can do that and the errors too. (Though I feel like we should lean on types as we only need one line of defence)

Reply:
We should rather get those from svelte. Maybe we can use omit?